src/main/java/gov/cabinetofice/gapuserservice/util/WebUtil.java

```diff
@@ -62,4 +64,13 @@ public static String parseUrlRequestParameters(String inputUrl, List<String> par
                     .concat(" ").concat(e.getMessage()));
         }
     }
-}
+
+    public static void validateRedirectUrl(final String inputUrl, final String baseUrl) throws MalformedURLException {
+        URL url = new URL(inputUrl);
+        String inputUrlHost = url.getProtocol() + "://" + url.getHost();
+        if (!baseUrl.startsWith(inputUrlHost)) {
+            log.error("Redirect url: " + inputUrl + " does not match host: ".concat(baseUrl));
+            throw new InvalidRequestException("Redirect url: " + inputUrl + " does not match host: ".concat(baseUrl));
```

Comment:
may as well use the `+` operator everywhere here? and define this string once:

```suggestion
            String errorMessage = "Redirect url: " + inputUrl + " does not match host: " + baseUrl;
            log.error(errorMessage);
            throw new InvalidRequestException(errorMessage);
```